app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    new_numbers = []
+    each do |i|
+      new_numbers.push(i) if yield(i)
+    end
+    new_numbers
+  end
+
+  def my_all?
+    condition = true
+    each do |j|
+      condition = false unless yield(j)
+    end
+    condition
+  end
+
+  def my_any?
+    each do |j|
+      return true if yield(j)
+    end
+    false
+  end
+
+  def my_none?
+    condition = true
+    each do |j|
+      condition = false if yield(j)
+    end
+    condition
+  end
+
+  def my_count(counted)
+    counting = 0
+    each do |j|
+      counting += 1 if j == counted
+    end
+    counting
+  end
+
+  def my_map(proc = nil)
+    arr = []
+    each do |i|
+      if !proc.nil?
+        arr.push(proc.call(i))
+      else
+        arr.push(yield i)
+      end
+    end
+    arr
+  end
+
+  def my_inject(arg = nil)
+    product = 0
+    i = 0
+    while i < length
+      product = yield product, self[i]
+      if product.zero?
+        product = 1
+        product = yield product, self[i]
+      end
+      i += 1
+    end
+    product = yield product, arg unless arg.nil?
+    product
+  end
+end
```

Comment:
1. When a block is given without an initial value should combine all elements of enum by applying a binary operation, specified by a block::range. For example:
`(1..5).my_inject { |prod, n| prod * n }` - should return `120`

2. When a block is given with an initial value should combine all elements of enum by applying a binary operation, specified by a block:
`(1..5).my_inject(2) { |prod, n| prod * n }` - should return `240`

3. When a symbol is specified without an initial value should combine each element of the collection by applying the symbol as a named method. For example:
`[1, 2, 3].my_inject(:+)` - should return `6`

4. When a symbol is specified with an initial value should combine each element of the collection by applying the symbol as a named method::range. For example:
`(1..5).my_inject(2, :*)` - should return `240`